### PR TITLE
Shader foundry/tests custom srp target

### PR DESCRIPTION
**Pull request by joshua-davis**
# **Please read the [Contributing guide](CONTRIBUTING.md) before making a PR.**

* Read the [Graphics repository & Yamato FAQ](http://go/graphics-yamato-faq).

---
### Purpose of this PR
https://jira.unity3d.com/browse/PLAT-1092
This change is adding a custom SRP and a custom target for the foundry integration tests. The purpose of this is two fold:
- Remove invalid dependencies on URP
- Fixup tests that couldn't easily test alpha usage due to how URP Unlit is written.

---
### Testing status
Verified that the tests worked

---
### Comments to reviewers
The majority of this change is the custom target. I did a reasonable pass on cleaning up the shader library from what URP would've used but it's still not tiny due to how the SRPs work.

The project ".asset" file changes are me just saving the project so I could get the custom SRP set. Some of the changes might not be necessary but I didn't want to narrow them down and it's nice to have the project in a cleaner state.

### CODE REVIEW

File: TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        {
+            m_Parent.Render(renderContext, cameras);
+        }
+    }
+
+    [ExecuteInEditMode]
+    public class SimpleSRPFoundryPipeline : RenderPipelineAsset
+    {
+#if UNITY_EDITOR
+        [UnityEditor.MenuItem("RenderPipeline/Create SimpleSRPFoundryPipeline")]
```
Comment (jessebarker) [inline TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs:27]: I think the `#if UNITY_EDITOR` is fine to have, but is there any chance that any of this code actually gets run in a player (i.e., do we actually _need_ it)?

**Review by jessebarker — APPROVED** [review]
LGTM

I suspect some of the hlsl could be reduced (one or two trivial things), but I don't think it's worth doing for a test project renderer.

File: TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs (hunk 1 above)
Comment (joshua-davis) [reply TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs:27]: This was copied from the SRP Batcher tests custom SRP.
This creates an menu option to make it easy to create a new asset file for the srp pipeline. The pipeline is then used as is, but the code to create a new asset isn't needed outside of the editor.

**Review by joravainen — APPROVED** [review]

File: TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+                if (!camera.TryGetCullingParameters(out cullingParams))
+                    continue;
+                CullingResults cull = context.Cull(ref cullingParams);
+
+                // Setup camera for rendering (sets render target, view/projection matrices and other
+                // per-camera built-in shader variables).
+                context.SetupCameraProperties(camera);
+
+                // render opaque objects using Deferred pass
```
Comment (cdxntchou) [inline TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs:61]: Forward not deferred, right?

**Review by cdxntchou — APPROVED** [review]
Looks good.

File: TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs (hunk 2 above)
Comment (joshua-davis) [reply TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs:61]: That is correct. That's what I get for copying from the SRP Batcher's custom SRP

File: TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs (hunk 1 above)
Comment (aleks01010101) [reply TestProjects/ShaderFoundry/Assets/CommonAssets/Editor/Targets/SRP/SimpleSRPFoundryPipeline.cs:27]: This is under "<>/Editor/<>" folder in the structure already, so I think it won't get included in the player anyway

**Review by aleks01010101 — APPROVED** [review]
LGTM!

**Pull request merged**